src/d3a/models/area/markets.py

```diff
@@ -15,84 +15,158 @@
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-from pendulum import DateTime # noqa
-from typing import Dict  # noqa
-
-from d3a.models.market import GridFee
-from d3a.models.market.two_sided import TwoSidedMarket
-from d3a.models.market.one_sided import OneSidedMarket
-from d3a.models.market.balancing import BalancingMarket
-from d3a.models.market import Market # noqa
-from d3a_interface.constants_limits import ConstSettings
 from collections import OrderedDict
+from typing import Dict, List, TYPE_CHECKING
+
+from d3a import constants
 from d3a.d3a_core.util import is_timeslot_in_simulation_duration
-import d3a.constants
+from d3a.models.market import GridFee, Market
+from d3a.models.market.balancing import BalancingMarket
+from d3a.models.market.one_sided import OneSidedMarket
+from d3a.models.market.two_sided import TwoSidedMarket
+from d3a_interface.constants_limits import ConstSettings, GlobalConfig
+from pendulum import DateTime
+
+if TYPE_CHECKING:
+    from d3a.models.area import Area
 
 
 class AreaMarkets:
+    """Class that holds all markets for areas."""
+
     def __init__(self, area_log):
-        # Children trade in `markets`
         self.log = area_log
+        # Children trade in `markets`
         self.markets = OrderedDict()  # type: Dict[DateTime, Market]
         self.balancing_markets = OrderedDict()  # type: Dict[DateTime, BalancingMarket]
         # Past markets
         self.past_markets = OrderedDict()  # type: Dict[DateTime, Market]
         self.past_balancing_markets = OrderedDict()  # type: Dict[DateTime, BalancingMarket]
-        self._indexed_future_markets = {}
+        # Settlement markets
+        self.settlement_markets = OrderedDict()  # type: Dict[DateTime, Market]
+        self.past_settlement_markets = OrderedDict()  # type: Dict[DateTime, Market]
+        self.indexed_future_markets = {}
```

Comment:
What's the `indexed_` prefix stating?
For example, is there a difference between `self.past_settlement_markets` and `self.indexed_future_markets`? If not, I would suggest to just name this variable as the others:
```suggestion
        self.future_markets = {}
```
(And maybe even using `OrderedDict()` as with the other ones.

Reply:
Indexed future markets is used to denote that it is an id-indexed dict. Reason why we added this was a performance improvement because the operation of searching a market by id was quite frequent in the futures markets and maintaining an indexed dict had a significant performance improvement. 
Hannes already answered in another comment that will be dealt with the new implementation of the futures market, which will migrate multiple futures markets to one, thus rendering this indexing obsolete.